lineapy/utils/logging_config.py

```diff
@@ -18,26 +18,42 @@
 
 
 def configure_logging(level=None, LOG_SQL=False):
-    # Get the loglevel from `LOGGING_ENV_VARIABLE` or set to WARNING
-    # if not defined
+    """
+    Configure logging for Lineapy.
+
+    Logging level is read first from the function parameter,
+    then lineapy_config options and defaults to INFO.
+
+    This function should be idempotent.
+    """
+
+    # Get the loglevel from options or set to INFO if not defined
     level = level or getattr(logging, options.logging_level, logging.INFO)
     # Disable black logging
     # https://github.com/psf/black/issues/2058
     logging.getLogger("blib2to3").setLevel(logging.ERROR)
 
+    # Set SQL log levels to match alembic.ini file
     if LOG_SQL:
-        logging.getLogger("sqlalchemy.engine").setLevel(logging.INFO)
-
-    logging.basicConfig(
```

Comment:
Clarifying comment on why we are removing this:

https://docs.python.org/3/library/logging.html#logging.basicConfig 

> This function does nothing if the root logger already has handlers configured, unless the keyword argument force is set to True.

This means that this line actually no-ops in our current code. 

I elected not to use force=True parameter which could have fixed this since this option was only introduced in python 3.8  and would make us not compatible with <3.8

Instead the chosen implementation is to set all of these configurations ourselves.